encoding/text/marshal.go

```diff
@@ -297,98 +312,174 @@ func codeOrderFields(s schema.Node_structNode) []schema.Field {
 	return fields
 }
 
+func (enc *Encoder) writeListStart() {
```

Comment:
These 3 helper functions make it harder to follow the logic below. Please inline them.

Reply:
Done.